db.go

```diff
@@ -127,48 +150,93 @@ func (x *dummyAuthenticator) ResetPasswordFinish(identity string, token string,
 	}
 }
 
-func (x *dummyAuthenticator) CreateIdentity(identity, password string) error {
+func (x *dummyUserStore) CreateIdentity(identity, password string) error {
 	x.usersLock.Lock()
 	defer x.usersLock.Unlock()
-	if _, exists := x.users[CanonicalizeIdentity(identity)]; !exists {
-		x.users[CanonicalizeIdentity(identity)] = &dummyUser{identity, password, ""}
+	user := getUser(x.users, identity)
+	if user == nil {
+		userId := generateUserId(x.users)
+		x.users[userId] = &dummyUser{userId, identity, password, ""}
 		return nil
 	} else {
 		return ErrIdentityExists
 	}
 }
 
-func (x *dummyAuthenticator) RenameIdentity(oldIdent, newIdent string) error {
+func (x *dummyUserStore) RenameIdentity(oldIdent, newIdent string) error {
 	x.usersLock.Lock()
 	defer x.usersLock.Unlock()
 
-	if _, exists := x.users[CanonicalizeIdentity(newIdent)]; exists {
-		return ErrIdentityExists
-	}
-
-	oldIdent = CanonicalizeIdentity(oldIdent)
 	newKey := CanonicalizeIdentity(newIdent)
+	oldIdent = CanonicalizeIdentity(oldIdent)
+	newUser := getUser(x.users, newKey)
+	if newUser == nil {
+		oldUser := getUser(x.users, oldIdent)
 
-	if user, exists := x.users[oldIdent]; exists {
-		delete(x.users, oldIdent)
-		x.users[newKey] = &dummyUser{newIdent, user.password, ""}
-		return nil
+		if oldUser != nil {
+			x.users[oldUser.userId].name = newIdent
+			return nil
+		} else {
+			return ErrIdentityAuthNotFound
+		}
 	} else {
-		return ErrIdentityAuthNotFound
+		return ErrIdentityExists
 	}
 }
 
-func (x *dummyAuthenticator) GetIdentities() ([]string, error) {
+func (x *dummyUserStore) GetIdentities() ([]string, error) {
 	x.usersLock.RLock()
 	list := []string{}
-	for identity, _ := range x.users {
-		list = append(list, identity)
+	for _, v := range x.users {
```

Comment:
Stick to using `defer unlock`, because it's consistent with all the other member functions.